packages/client/hmi-client/src/utils/petrinet-color-palette.ts

```diff
@@ -28,9 +43,23 @@ function setNodeTypeColor(ids: string[]): void {
 			domain.push(id);
 		}
 	});
-	scale = d3.scaleOrdinal(nodeTypeColors).domain(domain);
+	scaleNodeType = d3.scaleOrdinal(nodeTypeColors).domain(domain);
 }
 
 export function useNodeTypeColorPalette() {
 	return { getNodeTypeColor, setNodeTypeColor };
 }
+
+let scaleNestedType = d3.scaleLinear().domain([0, 2]);
+
+function getNestedTypeColor(id: number): string {
+	return d3.piecewise(d3.interpolateRgb.gamma(2.2), nestedTypeColors)(scaleNestedType(id));
+}
+
+function setNestedTypeColor(setDomain: [number, number]): void {
+	scaleNestedType = d3.scaleLinear().domain(setDomain);
+}
+
+export function useNestedTypeColorPalette() {
+	return { getNestedTypeColor, setNestedTypeColor };
+}
```

Comment:
I don't understand a couple of things here
- why are we interpolating colours, and
- why scaleNestedType a shared variable